lib/web/apiserver_test.go

```diff
@@ -1851,6 +1881,45 @@ func (s *WebSuite) waitForRawEvent(ws *websocket.Conn, timeout time.Duration) er
 	}
 }
 
+func (s *WebSuite) waitForCloseEvent(ws *websocket.Conn, timeout time.Duration) error {
+	timeoutContext, timeoutCancel := context.WithTimeout(context.Background(), timeout)
+	defer timeoutCancel()
+	doneContext, doneCancel := context.WithCancel(context.Background())
+	defer doneCancel()
+
+	go func() {
+		for {
+			time.Sleep(250 * time.Millisecond)
```

Comment:
The time for a full Teleport unit test run in CI is around 5 minutes at the moment - an extra second or two isn't going to make a huge difference. I'd rather have a slightly slow test that works than no test at all - just in case the planned refactoring never happens or takes longer than expected.

Reply:
Actually this loop doesn't need a sleep at all.
Read from a websocket should block.

I do agree with Gus. While slow tests are annoying we do suffer from low coverage more.

Reply:
This is not a general discussion about unit-tests which is obvious (so I agree with Gus too). This is about being mindful as in everything we do. Adding a dozens of lines of code and introducing additional delays to test not-critical 4 lines of code does not make much sense to me. 

@awly  yeah, that loop is not needed at all. I quickly checked other tests in this file which has the same unnecessary loop. It seems we can make our tests ~5sec faster.

Again, I would slightly refactor `streamTerminal` and fix its existing tests altogether rather than becoming a victim of a  "broken window" effect.

Reply:
> Adding a dozens of lines of code and introducing additional delays to test not-critical 4 lines of code does not make much sense to me.

I disagree with that philosophy.
The opposite extreme would be only testing critical parts of code (leaving out a ton of non-core functionality).
While writing smaller tests is definitely better, we shouldn't be reluctant to test code just because it takes a lot of lines.
Also, the number of lines being tested is not an important metric. Even 1 line can make a huge difference in behavior.

If the main issue is how long tests take to run, we need to move away from `gocheck` and use parallelized (`t.Parallel()`) standard go tests.

Reply:
> The opposite extreme

That's the thing that it's not about black/white situation or extremes which is the _main point_. Writing a good unit-test is a mindful process. It increases the overall complexity of the system as any other part of production code with the same consequences. 

Again, my choice would be to fix the function to make it testable than throwing yet another potentially flaky test. If we do not have time for this, I would rather drop the _expensive_ test if said functionality is not critical.

Reply:
> If the main issue is how long tests take to run, we need to move away from gocheck and use parallelized (t.Parallel()) standard go tests.

@awly  The main issue is that we are not testing it correctly. Most of these tests do not need that much time to run. `t.Parallel` will make it even worse if not addressed correctly.

Reply:
You provide some good points. I don't necessarily agree with everything (like the cost of writing a test or whether this specific test is correct), but that's okay.
We can debate software testing strategy separately elsewhere :)

To unblock this PR (and not take up more of @kimlisa's time), let's delete the test.
Refactoring the code to be more testable would take a while.

At some point in the future, I'd like to spend some dedicated time improving test coverage outside of bug fixes and feature work. Hopefully, this fix won't regress before then.

Reply:
reverted testing